### PR TITLE
NEW JOB: Blueshield Officer

### PR DESCRIPTION
**Pull request by swissloaf**
This adds the Blueshield officer to the game, a consequence of staff discussion over a couple of days

The Blueshield officer is responsible for the protection of all of the Command staff (with Captain being the priority). His office replaces the Command EVA room. Several new sprites have been added and most likely will be updated. You can view them in the sprite check below.

> Most of your time will be spent shadowing those you have been assigned to protect, making sure that they're alive and well, and generally staring at a crew monitor to make sure they're still on sensors. Your job is to keep everyone in Command, as well as CentCom VIPs, alive.
> 
> One very important detail of your job is that you are not Security. Do not let your Security (and Detective) access fool you. Your ID lets you reach every single Command Office on the station, not just the Head of Security. You should not be concerned with the goings-on at Security, nor should you be doing their job for them. You are not a Security Officer with a private office and snazzy outfit. You are a bodyguard.

- [x]  Plasmaman outfit
- [x] Regular outfit
- [x] Port outfits 
- [x] Exempt blueshield from all antags
- [x] Finish office on maps other than default map
- [ ] Additional locker or spawn resources
- [x] Solve issue of racism towards nonhumans for this particular role
- [x] figure out why job selection is fucked and wide
- [x] Blueshield Coat issue
- [x] HUDglasses icon **(icon uploaded, issue with setting icon)**
- [x] Make changes if its human only

Discussion on this PR should be limited towards the actual code aspects of it (i.e. the spawning equipment, locker equipment, etc..)

Discussion about the nature of the role is accepted however don't direct it all to me as I'm not the only person behind this.

image
(character sprite is shitty but just because it is large)

#### Changelog

:cl:  
rscadd: Adds a new job: Blueshield Officer 
/:cl:

### CODE REVIEW

**Comment by bowlofcereal** [comment]
yeah why not, you want rp this is rp

**Comment by Jarod1200** [comment]
Aight but where did the command space suits move to

**Comment by yacabo** [comment]
Sounds great but where's EVA going to go?

**Comment by swissloaf** [comment]
> Sounds great but where's EVA going to go?

EVA still exists it’s just the small area to the right of it that got replaced

**Comment by Marmio64** [comment]
that sprite is pretty cool actually

**Comment by MrHorizons** [comment]
> Aight but where did the command space suits move to

According to Swiss and some others, Command EVA was useless, so apparently it's been yeeted–on Box, at least.

**Comment by swissloaf** [comment]
> > Aight but where did the command space suits move to
> 
> According to Swiss and some others, Command EVA was useless, so apparently it's been yeeted–on Box, at least.

According to ynot****

**Comment by athena413** [comment]
all maps, please. i think this'll be important enough to be on omega too

**Comment by swissloaf** [comment]
> all maps, please. i think this'll be important enough to be on omega too

I plan to I just haven’t gotten to it yet

**Comment by jcatjmeow** [comment]
Just put the command eva suits in regular eva tbh they look cooler

**Comment by ToGWtF** [comment]
CE has a hardsuit in their office.
RD has a hardsuit in their office.
CMO has a hardsuit in their office.
Captain has a hardsuit in their office.
The only one who NEEDS command EVA is the HoP.

What if we just put a single, maybe the two if you really care, command EVA suit storage units in head of staff meeting room? Or on the bridge, in a wall-safe like how escape pods have emergency suits.

**Comment by TheLinuxOS** [comment]
I agree with the assessment that command EVA is useless, not once have I used it or seen it used. Also love the idea in general

**Comment by maxion12345** [comment]
Command EVA has mats and tools for command to use. For most , useless. But, say your a HOP who wants to modify your line. You got toolbelts, metal, glass, plasteel and all that good shit you need, without having to scavenge for tools. Especially useful during high pop when the 13 assistants have stripped tool storage empty. I'd prefer it more to be near the brig, honestly. I'll make a suggestion tomorrow for what id think would be a way to keep the head-only tool area/armored suits, and have the blue shield be on the sec hallway.

**Comment by ghost** [comment]
So basically a job for validhunters/pseudo sec, then.

Just like it is on every server.

You are going to regret adding this role.

**Review by JamieD1 — CHANGES_REQUESTED** [review]
Under HC Review due to Ongoing changes planned

File: code/modules/jobs/job_types/blueshield.dm
Comment (Tipy1802) [inline code/modules/jobs/job_types/blueshield.dm:23]: I feel like the blueshield would get paid by command, not sec

**Comment by patpol4** [comment]
Rules would have to be introduced first so this role cannot behave like a security officer with high access.

**Comment by Ratvar** [comment]
This will behave like assistant with guns and cap access instead, validing whatever threatens their metabuddy head and collecting departmental gamer loot. What exactly good does this add?

File: code/modules/jobs/job_types/blueshield.dm
Comment (swissloaf) [reply code/modules/jobs/job_types/blueshield.dm:17]: a result of quickwriting

Comment (swissloaf) [reply code/modules/jobs/job_types/blueshield.dm:17]: changes made

Comment (swissloaf) [reply code/modules/jobs/job_types/blueshield.dm:23]: There is no Command department to be paid by. The Captain is even paid by Security.

**Comment by ThatGuyPhil** [comment]
Ive seen this somewhere before...

**Comment by FLME** [comment]
image
this APC is broken, you put the wire knot on the same tile as where the APC actually is, not where the APC visually is.
image
do this instead

**Review by athena413 — CHANGES_REQUESTED** [review]
whoops an actual review
image
this is fucked, the wire knot needs to be on the same tile as the actual APC and not where the APC looks like it is, here's an example
image

**Comment by swissloaf** [comment]
> whoops an actual review
> image
> this is fucked, the wire knot needs to be on the same tile as the actual APC and not where the APC looks like it is, here's an example
> image

Ignore that, thats just a leftover from when I moved the office a block down and didnt remember to rewire it

File: code/modules/jobs/job_types/blueshield.dm
Comment (Tipy1802) [reply code/modules/jobs/job_types/blueshield.dm:23]: Ah I see, my bad

**Comment by Partheo** [comment]
I really don't think a bodyguard should have a fancy wooden office

What is this job's relationship with security and space law supposed to be like? Do they answer to the HoS or Warden? Can they enforce space law?

**Comment by Marmio64** [comment]
Blueshield should hopefully be a bodyguard only role that is not allowed to validhunt or enforce space law. Their only interaction with sec should be handing them cuffed people if they tried to attack command.

**Comment by Xoxeyos** [comment]
We should focus on fixing space law before we add more stuff to handle, I also don't see the point of this guy since security isn't staffed most of the time anyways; these guys are supposed to defend heads, not solve crimes.

**Comment by patpol4** [comment]
Agree, this 100% needs council to define the rules for the role before merging.

**Comment by swissloaf** [comment]
@yogstation13/council

**Comment by swissloaf** [comment]
@JamieD1 Code is finished and can be reviewed now. The only thing left to do is add additional locker and spawn resources (maybe spawn with a medkit) and edit some spawn loadout to make it better. Everything else is finished.

**Comment by ghost** [comment]
What will happen is the blueshield will follow the captain around, get gamer loot from science and security, and follow the Captain (or HoS) into combat any time there's an antag.

And then 'defend' them. Just like on every other server it's added.

This is a validhunt role and no amount of special rules will change that because they are obligated to guard HEADS OF STAFF, and at least TWO of the most important heads of staff on the station will be in combat all the fucking time.

**Comment by JamieD1** [comment]
Dont fucking ping council when we have not even decided if we want it yet. Jesus. @swissloaf

**Comment by swissloaf** [comment]
> Dont fucking ping council when we have not even decided if we want it yet. Jesus. @swissloaf

Chill out, calm down, and take it easy. It's just a ping, you'll live. The ping was for council to see about discussing aspects of it and then a vote if wanted.

**Comment by Reed0506** [comment]
what if

security guard the heads if need be

instead of

whatever this is

File: code/game/objects/items/devices/radio/encryptionkey.dm
Comment (Hopekz) [inline code/game/objects/items/devices/radio/encryptionkey.dm]:
```suggestion
	name = "\improper Blueshield encryption key"
```

File: code/modules/jobs/job_types/blueshield.dm
Comment (Hopekz) [inline code/modules/jobs/job_types/blueshield.dm:42]:
```suggestion
```
So they get a disabler in their locker AND an energy gun?

**Review by Hopekz — CHANGES_REQUESTED** [review]

**Comment by Tipy1802** [comment]
> what if

>security guard the heads if need be

>instead of

>whatever this is

Well they don’t do that, they have no way to communicate with the heads over radio and are far too busy dealing with greytiders or antags

**Comment by Tipy1802** [comment]
> What will happen is the blueshield will follow the captain around, get gamer loot from science and security, and follow the Captain (or HoS) into combat any time there's an antag.

>And then 'defend' them. Just like on every other server it's added.

>This is a validhunt role and no amount of special rules will change that because they are obligated to guard HEADS OF STAFF, and at least TWO of the most important heads of staff on the station will be in combat all the fucking time.

Heads of staff are actually very defenceless considering Theos made their batons worthless. The captain can easily be killed and once that’s done the traitor gets a free win because he can steal his objectives no problem and loot the armoury

**Comment by ghost** [comment]
@Tipy1802 

>captain defenseless

ha haha ha ha ha ha ha ha ha ha ha ha

>batons worthless

ha ha ha ha ha ha ha ha ha

**Comment by Altoids1** [comment]
### I'm sorry Swiss, I can't say that I like this.

In my opinion **this is low-RP-as-design.** The greatest objective of the Security department, held mutually by everyone in it, is to protect the lives and valuables of the critical heads of the station from spessthreats. Delegating that work into one person, who just sits around next to the heads waiting for shit to happen, is just creating a shitsec-captain almost by design.

People seem to have realized that and are suggesting council create rules. When making code that could be a pain to admin, it's good to at least have a basic concept of what the new rules for this would be. Unfortunately, I don't see what rules they can exactly make. Forcing the blueshield to always guard a head means that they'll just validhunt along with the Captain or HoS all the time, while the CMO or whoever dies alone in aft maint.

Roles like this have been implemented on other servers, and I've yet to hear anyone here say that they like those roles on those servers. I'm very skeptical of taking big game design shifts like this by face-value, when we can have real empirical evidence about its quality and enjoyability.

This has the potential to alter a lot about how antags work, how difficult they are, and how much margin they have to do funnier things and dick around instead of needing to go into a tryhard sweat to resolve the constant 1-2 people always surrounding the Captain. I can't say that I'm for it in any way.

File: code/modules/jobs/job_types/blueshield.dm
Comment (swissloaf) [reply code/modules/jobs/job_types/blueshield.dm:42]: What, you want them just to have a disabler?

Comment (Hopekz) [reply code/modules/jobs/job_types/blueshield.dm:42]: If you're going to say this is intentional at least add them to the plasma man backpack as well otherwise it looks like an oversight.

Comment (swissloaf) [reply code/modules/jobs/job_types/blueshield.dm:42]: Plasmaman gets it though

**Comment by patpol4** [comment]
Some maintainer/HC ping me if we actually decide to do this so i don't do the map review for no reason

**Comment by CMOisLing** [comment]
Cool idea but I think adams right on this one

**Review by Partheo — APPROVED** [review]

**Comment by Tipy1802** [comment]
I mean yes if the captain gets attacked by a traitor the blueshield will have to defend the captain, I don't see what's the problem in there

**Comment by Xoxeyos** [comment]
Yeah Blueshield is retarded, I really do hope this isn't merged.

**Comment by neeshacark** [comment]
> Yeah Blueshield is retarded, I really do hope this isn't merged.

im of the opposite opinion, merge please

**Comment by neeshacark** [comment]
so I have been thinking of a way to get around the current issue I do have with this.

have a rule that officer must switch what head of staff he is guarding every x minutes, this will give time for antags to assassinate their target should they be a head

a second rule: dont guard the hos, he is his own guard as he is very capable of defending himself

and, not a big code man but if possible, which i think it should be, make it so blueshield only is able to be played if there are atleast 2 heads of staff. this is to support the prior suggestion. one of those heads cant be the hos, for the prior reason stated in the second rule suggestion

**Comment by patpol4** [comment]
> so I have been thinking of a way to get around the current issue I do have with this.
> 
> have a rule that officer must switch what head of staff he is guarding every x minutes, this will give time for antags to assassinate their target should they be a head
> 
> a second rule: dont guard the hos, he is his own guard as he is very capable of defending himself


this is impossible to enforce by admins

**Comment by MegaEmpirical** [comment]
> In my opinion **this is low-RP-as-design.** The greatest objective of the Security department, held mutually by everyone in it, is to protect the lives and valuables of the critical heads of the station from spessthreats. Delegating that work into one person, who just sits around next to the heads waiting for shit to happen, is just creating a shitsec-captain almost by design.

The greatest job of Sec is keeping the greytide in line. 


> This has the potential to alter a lot about how antags work, how difficult they are, and how much margin they have to do funnier things and dick around instead of needing to go into a tryhard sweat to resolve the constant 1-2 people always surrounding the Captain. I can't say that I'm for it in any way.

I have no objections to the Captain having a guard given how often and easily they're taken by surprise for all access or to supplant captainship.

**Comment by ghost** [comment]
I really think this only makes sense if MRP is moved along more and the captain's role is specifically discouraged from engaging in hunting antags and engaging in security stuff like arrests.

**Comment by JackKindred** [comment]
I like it, more often than not, Sec's far too fed up with like, ten thousand other things happening all around to really 'cater' to the Heads most of the time, while sure, they can just use Common and scream bloody murder if an antag bad man toolboxes them, having a Blueshield officer around would increase the chances of survival of the only people NT _really_ cares about stationside, so I think it makes sense for them to issue the heads their own personal sec man. Plus, if you know what you're doing and you're competent in the slightest, taking out a head+Blueshield isn't too hard.

**Comment by Moja10** [comment]
I will say, if Blueshield is successful enough, the Captain's ability to validhunt or act as another member of sec might legitimately be removed in favor of having blueshield protection. But we will see, there has been plenty of discussion about this role and it won't simply be implemented as Captain Simp who is with the Captain gunning down valids in maint.

**Comment by neeshacark** [comment]
> I will say, if Blueshield is successful enough, the Captain's ability to validhunt or act as another member of sec might legitimately be removed in favor of having blueshield protection. But we will see, there has been plenty of discussion about this role and it won't simply be implemented as Captain Simp who is with the Captain gunning down valids in maint.

Yeah this was something I was thinking about too, albeit in a slightly different way.  What if instead of just serving as a bodyguard, the BO also serves as... a caretaker?  Instead of rushing into maints to help the captain shoot dat valid, part of their job could be making sure the captain isn't running into maint to begin with.  Problem with that is some captains would be all 'reEEeeE perma the blueshield he stunbatonned me and the tot got away REEeeeEEEe'

**Comment by patpol4** [comment]
Is this going anywhere?

**Comment by swissloaf** [comment]
> Is this going anywhere?

HCs havent decided if they want this yet

**Comment by alexkar598** [comment]
HC*

**Comment by swissloaf** [comment]
I’d been over a month

**Comment by neeshacark** [comment]
im still holding out hope

**Comment by JamieD1** [comment]
Closed until we have an idea of what we are doing with this.

**Comment by Tipy1802** [comment]
Why is this taking so long? Is really a role that exists on 90% of servers that all it does is follow the captain around. The concerns I have seen are that it would be a validhunting role but that's just no true and admins will ban any blueshield that hunts antagonists without any heads being in danger. Wouldn't it be better to test-merge it and then decide wether or not to merge it or not based on the results? I am tired of how every slightly experimental PR (that's not a map PR or a TG port) being halted by absured ammount of red tape

**Comment by patpol4** [comment]
> Why is this taking so long?

council

**Comment by swissloaf** [comment]
> > Why is this taking so long?
> 
> council

council have approved it i think

**Comment by ghost** [comment]
@Tipy1802 because it has conflicts

**Comment by Tipy1802** [comment]
> @Tipy1802 because it has conflicts

It has conflicts because Jamie said he won't merge it until... something...? IDK he keeps changing what he is waiting for

**Comment by BigBigga** [comment]
+1

**Comment by JamieD1** [comment]
Closed until we have an idea of what we are doing with this.

**Pull request closed without merging**
